### ISSUE TITLE
Pixel Normalization tensorflow impliment

### ISSUE THREAD

**Issue by kyrie2to11** [zhoumumu/VapSR#4]
您好，最近尝试用tf复现您的工作，尝试通过method:  tf.keras.GroupNormalization() 实现。有以下两个疑问

 1.  对于[B,  H, W, C ]形状的输入, pxiel norm 计算了多少次均值和方差呢， B x H x W次还是 H x W 次？ 结合您的paper和code，我觉得是前者，也就是不同照片同一位置的像素分别计算了不同的均值和方差。 请问是这样吗？
 
 2. tf官方API没看懂axis参数指定多个dim到底是啥意思 -> [API文档地址](https://www.tensorflow.org/api_docs/python/tf/keras/layers/GroupNormalization#args)，试了下面两种都不对 :( ，有空的话帮看下吧
     - tf.keras.GroupNormalization(groups=d_model,  axis=-1, epsilon=1e-8,) # 默认输入 x [B , H , W , C]
     - tf.keras.GroupNormalization(groups=B * H * W,  axis= （0，1，2）, epsilon=1e-8,) # 默认输入 x [B , H , W , C]
    
谢谢 ！

**Comment by zhoumumu** [comment]
抱歉抱歉回复晚啦，最近事情很多~~

1.H*W*N次，参见图3，LayerNorm N次，BatchNorm C次，InstanceNorm C*N次。

2.如何理解axis参数：normalize across的意思就是在哪个维度上做那个算均值方差的操作。一般来说标准的GroupNorm都是在Channel维度上做的，对应默认输入格式的最后一维，即默认值-1。那在Channel维度上做是什么意思呢？如果我在图3给你补个GroupNorm的示意图的话，它会是介于LayerNorm和InstanceNorm之间的情况，比如C上分两组那么C*H*W这个面的小方格要涂紫一半，计算次数为G*N次。（参考[torch文档的几个example](https://pytorch.org/docs/stable/generated/torch.nn.GroupNorm.html)）当你不想在Channel维度上分组了，可以用axis指定其他维。

我认为（没做实验哈）PixelNorm的一种正确的tf实现是：tf.keras.GroupNormalization(groups=H * W, axis= (1, 2))
然后你不要像我用LayerNorm实现一样，输入前后要转格式了，你这个直接[BHWC]输入就行。
（形象点来说原本axis设在C上，“分组的刀”在图3里就在C这条轴上竖着切，这里设axis等于(1,2)就在H*W这条轴上横着片）

比花~(✿◕‿◕✿)
